### PR TITLE
#57 Implement endpoints (backend) to create a dossier of a user.

### LINKED ISSUE #57: Setup endpoints to create a dossier for a user

### PR DESCRIPTION
**Pull request by zihao429**
Set up endpoint for initiators to create a new dossier, to which change requests may be added.
This PR closes #57

### CODE REVIEW

**Comment by JamesPartsafas** [comment]
Everything looks good, but can you write tests for the DossierService in addition to the repository tests? (To cover the branches through the dossier creation function)

File: Server/ConcordiaCurriculumManager/Controllers/DossierController.cs [hunk 1]
```diff
@@ -51,4 +57,36 @@ public DossierController(IMapper mapper, ILogger<DossierController> logger, IUse
                 statusCode: StatusCodes.Status500InternalServerError);
         }
     }
+
+    [HttpPost(nameof(CreateDossierForUser))]
+    [SwaggerResponse(StatusCodes.Status400BadRequest, "Invalid input")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    [SwaggerResponse(StatusCodes.Status201Created, "Dossier created successfully", typeof(Guid))]
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/Controllers/DossierController.cs:64]: There's a mistake here, you don't return a typeof(GUID), you return a typeof(DossierDTO)

Comment (zihao429) [reply Server/ConcordiaCurriculumManager/Controllers/DossierController.cs]: Sounds good.

**Comment by zihao429** [comment]
> Everything looks good, but can you write tests for the DossierService in addition to the repository tests? (To cover the branches through the dossier creation function)

Sounds good

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**